agreement_dynamic/models/header.py

```diff
@@ -0,0 +1,47 @@
+class Header:
+    def __init__(self):
+        self.header1_ = 0
+        self.header2_ = 0
+        self.header3_ = 0
+
+    @property
+    def header1(self):
+        return self.header1_
+
+    @property
+    def header2(self):
+        return self.header2_
```

Comment:
Maybe you can put this class to be as just one level.

Then in variables you can pass:

```
h1 = Header()
h2 = Header()
h3 = Header()
variables = {'h1': h1, 'h2', h2, ...}
```

And then in the template it will not be a big string to call, but just `h1.value`, `h1.next` and `h1.reset`

Even you can remove value by making

```
def __repr__(self):
    return self.value
```

and then it can just be: `$(h1).$(h2).$(h3.next): Some text`

`(h1).$(h2.next).$(h3.reset): Some other text`

It's still not ideal, could probably be implemented in a better way but it's already quite usable I think 

As a later improvement we could find a way of skipping the need for `reset` if we somehow make one header know about his child, so he can "pass down" the reset command to its child, which will in turn pass it down etc